### PR TITLE
fix: Enable multi widget selection in Anvil

### PR DESCRIPTION
**Pull request by marks0351**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
## Description
Fixing Selection to allow multi selection of widgets.
>
#### PR fixes following issue(s)
Fixes # (issue number)
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
	- Enhanced widget selection and interaction by updating event handling across components.
	- Modified types to align with event handling needs.
	- Updated event detail objects and handling in various hooks and utilities.
- **Tests**
	- Added new test cases for widget selection with Ctrl + Click.
	- Updated existing test cases for improved coverage and accuracy.
	- Enhanced test case descriptions for better clarity.
	- Introduced a new parameter in a test function for improved testing capabilities.

### CODE REVIEW

File: app/client/src/layoutSystems/anvil/integrations/sagas/anvilWidgetSelectionSaga.ts [hunk 1]
```diff
@@ -73,6 +71,7 @@ export function* selectAnvilWidget(
       type === SelectionRequestType.ShiftSelect
     ) {
       e.stopPropagation();
+      clickEvent.stopPropagation();
```
[the remark this answers is not in the document]
Comment (marks0351) [reply app/client/src/layoutSystems/anvil/integrations/sagas/anvilWidgetSelectionSaga.ts:74]: makes sense.

[the remark this answers is not in the document]
Comment (marks0351) [reply app/client/src/layoutSystems/anvil/integrations/sagas/anvilWidgetSelectionSaga.ts]: removed this changes, coz its was unnecessary, we only want the custom event to not propogate and not the click.

[the remark this answers is not in the document]
Comment (marks0351) [reply app/client/src/layoutSystems/anvil/integrations/sagas/anvilWidgetSelectionSaga.ts]: Thanks for pointing this out.

File: app/client/src/layoutSystems/anvil/integrations/sagas/anvilWidgetSelectionSaga.ts [hunk 2]
```diff
@@ -47,9 +45,9 @@ export function* selectAnvilWidget(
     !isWidgetAlreadySelected
   ) {
     let type: SelectionRequestType = SelectionRequestType.One;
-    if (e.metaKey || e.ctrlKey || (layoutSystemType && e.shiftKey)) {
+    if (clickEvent.metaKey || clickEvent.ctrlKey) {
```
[the remark this answers is not in the document]
Comment (riodeuno) [reply app/client/src/layoutSystems/anvil/integrations/sagas/anvilWidgetSelectionSaga.ts:48]: Yes, it is about a pattern. The pattern here is harder to debug. It is useful to only pass around the minimum data necessary.

**Review by riodeuno — APPROVED** [review]

**Pull request merged**